lib/testing.dart

```diff
@@ -10,14 +11,25 @@ import 'package:bazel_worker/bazel_worker.dart';
 export 'src/utils.dart' show protoToDelimitedBuffer;
 
 /// A [Stdin] mock object.
+///
+/// Note: When using this with an [AsyncWorkerLoop] you must call [close] in
+/// order for the loop to exit properly.
 class TestStdinStream implements Stdin {
+  /// Pending bytes to be delivered synchronously.
   final Queue<int> pendingBytes = new Queue<int>();
 
-  // Adds all the input bytes to this stream.
+  /// Controls the stream for async delivery of bytes.
+  final StreamController _controller = new StreamController();
+
+  /// Adds all the [bytes] to this stream.
   void addInputBytes(List<int> bytes) {
     pendingBytes.addAll(bytes);
+    _controller.add(bytes);
```

Comment:
I'm confused--is this class meant to be used synchronously, asynchronously, or both?  It seems strange to be sending the input bytes to two places.

Suggestion: have two mock objects, one for sync and one for async.